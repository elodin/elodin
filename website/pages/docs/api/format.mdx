import DocLayout from '../../../components/DocLayout.js'

export default DocLayout

# Format

```bash
yarn add --dev @elodin/format
```

Format beautifies a string of Elodin code in an opinionated approach.

## format

It exposes a single `format` method that just takes an single string of code.

### Example

```javascript
import { format } from '@elodin/format'

const input = `
style Button { backgroundColor: red
      paddingLeft: 10
paddingRight  :   10
}
`

const output = format(input)

output = `
style Button {
  backgroundColor: red
  paddingLeft: 10
  paddingRight: 10
}
`
```
