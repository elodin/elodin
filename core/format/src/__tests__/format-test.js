import format from '../format'

const file = `
# foo
variant Type { Primary }
variant Mode { Dark 

 Light}
style Flex {
   flexGrow: 1
   #foobar
      # baz
  flexDirection: row
}
style Button{justifyContent: flexStart #baz


      __borderWidth: .3
       backgroundColor: rgb(  255  200   155  )
#bam
  [ Type =   Primary  ] {  #bazbar 
    paddingLeft: 20[Mode=Dark]{ backgroundColor:green #bam 
      __paddingLeft: 10}}
  }

  style    Label {
lineHeight:   $lineHeight
    fontSize: 20

    color: rgb(200 120 0)
     [@hover] {
       color: rgba(255 0 120 percentage(50))
     }

     [@minWidth=320] {
       color:rgba(add(10 20) 0 20 percentage($percent)) [@hover] {
         color: green
       }
     }
}`

describe('Formatting elodin syntax', () => {
  it('should return beautiful syntax', () => {
    expect(format(file).code).toMatchSnapshot()
  })

  it('should return beautiful syntax with custom syntax', () => {
    expect(
      format(
        file,
        {
          ident: '    ',
        }.code
      )
    ).toMatchSnapshot()
  })
})
