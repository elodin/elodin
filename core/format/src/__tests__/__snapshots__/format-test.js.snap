// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Formatting elodin syntax should return beautiful syntax 1`] = `
"# foo
variant Type {
  Primary
}

variant Mode {
  Dark
  Light
}

style Flex {
  flexGrow: 1
  # foobar
  # baz
  flexDirection: row
}

style Button {
  # baz
  __borderWidth: .3
  justifyContent: flexStart
  backgroundColor: rgb(255 200 155)
  # bam
  [Type=Primary] {
    # bazbar
    paddingLeft: 20
    [Mode=Dark] {
      # bam
      __paddingLeft: 10
      backgroundColor: green
    }
  }
}

style Label {
  lineHeight: $lineHeight
  fontSize: 20
  color: rgb(200 120 0)
  [@hover] {
    color: rgba(255 0 120 percentage(50))
  }
  [@minWidth=320] {
    color: rgba(add(10 20) 0 20 percentage($percent))
    [@hover] {
      color: green
    }
  }
}"
`;

exports[`Formatting elodin syntax should return beautiful syntax with custom syntax 1`] = `
Object {
  "code": "# foo
variant Type {
  Primary
}

variant Mode {
  Dark
  Light
}

style Flex {
  flexGrow: 1
  # foobar
  # baz
  flexDirection: row
}

style Button {
  # baz
  __borderWidth: .3
  justifyContent: flexStart
  backgroundColor: rgb(255 200 155)
  # bam
  [Type=Primary] {
    # bazbar
    paddingLeft: 20
    [Mode=Dark] {
      # bam
      __paddingLeft: 10
      backgroundColor: green
    }
  }
}

style Label {
  lineHeight: $lineHeight
  fontSize: 20
  color: rgb(200 120 0)
  [@hover] {
    color: rgba(255 0 120 percentage(50))
  }
  [@minWidth=320] {
    color: rgba(add(10 20) 0 20 percentage($percent))
    [@hover] {
      color: green
    }
  }
}",
  "errors": Array [],
}
`;
